Seohee/Lv2/괄호 변환.js

```diff
@@ -0,0 +1,54 @@
+// 올바른 괄호 문자열인지 체크
+function isCorrectBracket(str) {
+    const stack = [];
+    let open = 0;
+    let close = 0;
+    for (let i = 0; i < str.length; i++) {
+        if (str[i] === '(') {
+            stack.push('(');
+            open++;
+        } else {
+            stack.pop();
+            close++;
+        }
+    }
+    if (stack.length === 0 && open === close)
+        return true;
+    else
+        return false;
+}
+function solution(p) {
+    // 괄호 열고, 닫기 구분
+    let open = 0;
+    let close = 0;
+    // 빈 문자열 -> 재귀 종료 1
+    if (!p) {
+        return '';
+    }
+
+    for (let i = 0; i < p.length; i++) {
+        // 괄호 개수 파악하기
+        if (p[i] === '(') open += 1;
+        else close += 1;
+        // 열고 닫는 괄호가 같은 횟수로 등장했을때, 
+        if (open === close) {
+            // u가 올바른 괄호 문자열 일때, 3
+            if (isCorrectBracket(p.slice(0, i + 1))) {
+                // 3 - 1
+                return p.slice(0, i + 1) + solution(p.slice(i + 1));
+            } else {
+                // u가 올바른 괄호 문자열이 아닐때,
+                // 4 -1 ~ 3
+                let answer = '(' + solution(p.slice(i + 1)) + ')';
+                // 4 - 4
+                for (let j = 1; j < i; j++) {
+                    if (p[j] === '(') answer += ')';
+                    else answer += '(';
+                }
+                // 4-5
+                return answer;
+            }
```

Comment:
```suggestion
            } 
                // u가 올바른 괄호 문자열이 아닐때,
                // 4 -1 ~ 3
                let answer = '(' + solution(p.slice(i + 1)) + ')';
                // 4 - 4
                for (let j = 1; j < i; j++) {
                    if (p[j] === '(') answer += ')';
                    else answer += '(';
                }
                // 4-5
                return answer;
```
if문 내에서 무조건적으로 return이 된다면 else라는 키워드를 없이 코드를 작성해도 결과가 같고 들여쓰기의 깊이가 줄어들기 때문에 가독성이 좋아져요.
no-else-return: https://eslint.org/docs/rules/no-else-return